Source/DafnyCore/Compilers/Dafny/AST.dfy

```diff
@@ -70,9 +75,9 @@ module {:extern "DAST"} DAST {
     Neq(referential: bool, nullable: bool) |
     Div() | EuclidianDiv() |
     Mod() | EuclidianMod() |
-    Implies() |
+    Implies() | // TODO: REplace by Not Or
```

Comment:
That seems like a good idea to me, but it assumes that the goal of the DAST is to be as simple as possible, rather than retaining extra information that might produce better output in some contexts. I think that's a good design goal personally, but this file should be more explicit about documenting that.

Reply:
Ideally I want:
- The Compiled Dafny AST to be minimal
- We want the generated code to look idiomatic and close to the original Dafny file if possible.

I was able to accomplish both goals the following way, as this step has already been accomplished in the next PR. The design is the following:
1) We keep only the minimum amount of operators (so `BinaryOp(Implies, x, y)` becomes `BinaryOp(Or, Not(x), y)`, same for all the variants of <= and < and != and == etc.
2) I added a "format" attribute to `BinaryOp` and `UnaryOp` so that compilers can decide to enhance the rendering of the operators based on original Dafny information, but that should only be an optional idiomatic rendering step or a transformation of the target language AST to another more idiomatic target language AST.
This ensures conversions are done minimally .

Reply:
> Ideally I want:
> * The Compiled Dafny AST to be minimal
> * We want the generated code to look idiomatic and close to the original Dafny file if possible.

Both good goals to have, but there will be times when they conflict. :)

Sounds like you're saying in this case we might annotate the AST further if we decided we need to know when the original Dafny source was `A ==> B` as opposed to `!A || B`. I think that's a reasonable general approach: keep the AST itself minimal so getting to simple correct translation is easier, but add more additional information when you feel the need to improve the output.

I think it's worth stating that tenet in or around this file somewhere.

Reply:
I just added to the top of Dafny-compiler-rust.dfy for the next PR:

```dafny
// Dafny to Rust compilation tenets:
// - The Compiled Dafny AST should be minimal
// - The generated code should look idiomatic and close to the original Dafny file if possible
```